basis/listener/listener.factor

```diff
@@ -24,8 +24,13 @@ H{
 M: object prompt.
     nip prompt-style get-global format bl flush ;
 
+: with-ctrl-break ( quot -- )
+    enable-ctrl-break
+    [ disable-ctrl-break ] [ ] cleanup ; inline
+
 : parse-lines-interactive ( lines -- quot/f )
-    [ parse-lines ] with-compilation-unit ;
+    [ [ parse-lines ] with-ctrl-break
```

Comment:
@mrjbq7: I've added a patch to this PR that makes enable-ctrl-break calls opt-in, off by default.

Will you merge now?